core/src/replay_stage.rs

```diff
@@ -1120,6 +1121,7 @@ impl ReplayStage {
             next_slots.sort();
             next_slots
         });
+        let mut new_banks = Vec::new();
         for (parent_slot, children) in next_slots {
             let parent_bank = frozen_banks
                 .get(&parent_slot)
```

Comment:
Does `forks.get(child_slot)` below need to reference new_banks as well?

Reply:
I think we can actually assert there that new_banks doesn't ccontain cchild slot, a child should only have one parent

Reply:
ok. so `assert!(new_banks.get(&child_slot).is_none());`?